src/mds/mdstypes.h

```diff
@@ -75,10 +75,46 @@
 
 
 typedef int32_t mds_rank_t;
+typedef int32_t mds_namespace_t;
+
+
+
 BOOST_STRONG_TYPEDEF(uint64_t, mds_gid_t)
 extern const mds_gid_t MDS_GID_NONE;
+extern const mds_namespace_t MDS_NAMESPACE_NONE;
+// The namespace ID of the anonymous default filesystem from legacy systems
+extern const mds_namespace_t MDS_NAMESPACE_ANONYMOUS;
 extern const mds_rank_t MDS_RANK_NONE;
```

Comment:
If we go c++11 the extern const can instead be

 constexpr mds_namespace_t MDS_NAMESPACE_NONE = {0};

etc.  That avoids having to define it in a .cc file somewhere and lets the compiler sort it out at build time.